### PR TITLE
[chaos] migrate tests to civ2.

### PR DESCRIPTION
**Pull request by aslonnie**
using host network rather than dind

### CODE REVIEW

**Comment by aslonnie** [comment]
pledge to change the many jobs test to only submit 50 jobs rather than 100. I feel that it does not really need to run for 50mins to detect possible errors. (it will be run on every single commit postmerge).

probably 30 will also be enough..

File: ci/k8s/run-chaos-test.sh [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+bash python/ray/tests/chaos/prepare_env.sh
+
+if [[ "${CHAOS_FAULT}" != "no_fault" ]]; then
+    kubectl apply -f "python/ray/tests/chaos/${CHAOS_FAULT}.yaml"
+fi
+
+echo "--- Prepare the test container."
+docker run -d --network host --name chaos-test ray-ci:kuberay-test sleep infinity
+docker exec -ti chaos-test mkdir -p /tmp/rayci/python/ray/tests/chaos
```
Comment (can-anyscale) [inline ci/k8s/run-chaos-test.sh:20]: does the script have to run inside this folder path? what if just copy the files in to /tmp/rayci

**Review by can-anyscale — COMMENTED** [review]
Review mostly on the code structure, not the test logic; if we can get test owners to review that's great; otherwise i can stamp early next week

**Review by rynewang — APPROVED** [review]

File: ci/k8s/run-chaos-test.sh (hunk 1 above)
Comment (aslonnie) [reply ci/k8s/run-chaos-test.sh:20]: the script's content has assumption that is runs at the root. it is probably very easy to change, but I do not want to change it here.

if you feel strongly, I can do it in a follow up.

only files under the directory is copied in, so the dependency is well restricted.

**Comment by aslonnie** [comment]
test owner already approved.

File: .buildkite/k8s.rayci.yml [hunk 2]
```diff
@@ -0,0 +1,25 @@
+group: "k8s tests"
```
[the remark this answers is not in the document]
Comment (aslonnie) [reply .buildkite/k8s.rayci.yml:1]: done.

the chaos test actually does not test kuberay but just uses kuberay as a convenient way to run ray on k8s. it needs k8s for the easy components to inject network faults or delays.

File: ci/k8s/kind-dind.config.yaml [hunk 3]
```diff
@@ -0,0 +1,17 @@
+kind: Cluster
```
[the remark this answers is not in the document]
Comment (aslonnie) [reply ci/k8s/kind-dind.config.yaml:1]: done. this file will be deleted very soon, after the operator test is migrated.

File: ci/k8s/kind.config.yaml [hunk 4]
```diff
@@ -1,17 +1,7 @@
 kind: Cluster
```
[the remark this answers is not in the document]
Comment (aslonnie) [reply ci/k8s/kind.config.yaml:1]: done.

**Review by can-anyscale — APPROVED** [review]

**Pull request merged**